src/redux/Missions/index.js

```diff
@@ -1,20 +1,52 @@
 // action types
-export const TEST = 'TEST';
+const JOIN_MISSIONS = 'JOIN_MISSIONS';
+const LEAVE_MISSIONS = 'LEAVE_MISSIONS';
+const UPDATE_MISSIONS = 'UPDATE_MISSIONS';
 
 // state
 const initialState = [];
 
 // action creators
-export const test = () => ({
-  type: TEST,
+export const join = (id) => async (dispatch) => dispatch({
+  type: JOIN_MISSIONS,
+  payload: id,
 });
+export const leave = (id) => async (dispatch) => dispatch({
+  type: LEAVE_MISSIONS,
+  payload: id,
+});
+
+export const showMissions = () => async (dispatch) => {
```

Comment:
actionCreatore are meant to be pure functions so an asyncThunk function is more suitable here